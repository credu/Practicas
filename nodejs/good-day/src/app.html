<!DOCTYPE html>
<html lang="es">
    <head>
        <meta charset="UTF-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <title>QR code</title>
        <link rel="shortcut icon" type="image/png" href="https://web.whatsapp.com/img/favicon_c5088e888c97ad440a61d247596f88e5.png" src="https://web.whatsapp.com/img/favicon_c5088e888c97ad440a61d247596f88e5.png">
        <style>
        * { 
            margin: 0;
            padding: 0;
            user-select: none;
            box-sizing: border-box;
            -moz-user-select: none; 
            -webkit-user-select: none;
            -ms-user-select: none;
        }
        body {
            background-color: #111B21;
            font-family: Segoe UI,Helvetica Neue,Helvetica,Lucida Grande,Arial,Ubuntu,Cantarell,Fira Sans,sans-serif;
        }
        nav {
            background-color: #00A884;
            width: 100%;
            padding-bottom: 2%;
        }

        nav a {
            margin: 0 auto;
            display: block;
            width: 100%;
            max-width: 1000px;
            color:#fff;
            text-decoration: none;
            font-weight: 510;
            line-height: 2.8;
        }

        body::-webkit-scrollbar {
            width: 6px;
            background: #111B21;
        }
        body::-webkit-scrollbar-thumb {
            background: #313537;
            border-radius: 1px;
        }

        .container {
            display: flex;
            flex-wrap: wrap;
            background-color: #fff;
            border-radius: 3px;
            box-shadow: 0 17px 50px 0 rgb(0 0 0 / 19%), 0 12px 15px 0 rgb(0 0 0 / 24%);
            margin: 0 auto;
            margin-top: -2%;
            padding: 40px 25px;
            width: 100%;
            max-width: 1000px;
        }

        .container > * {
            flex: 2 1 0;
        }

        ul {
            list-style: decimal;
            margin-left: 25px;
        }

        li {
            color: #4a4a4a;
            font-size: 20px;
            line-height: 28px;
            margin-top: 2.9%;
        }

        h2 {
            font-size: 28px;
            font-weight: 300;
            line-height: normal;
            color: #525252;
        }

        .container-qr {
            flex: 1 1 0;
            display: flex;
            justify-content: center;
            align-items: center;
            position: relative;
            aspect-ratio: 1/1;
        }

        .container-qr svg {
            position: absolute;
        }


    </style>
</head>
<body>
    <nav>
        <a href="/">GOOD DAY APP</a>
    </nav>
    <main>
        <section class="container" >
            <div>
                <h2>Para usar WhatsApp en tu computadora:</h2>
                <ul>
                    <li>Abre WhatsApp en tu teléfono</li>
                    <li>Toca Menú o Configuración y selecciona WhatsApp Web</li>
                    <li>Cuando se active la cámara, apunta tu teléfono hacia esta pantalla para escanear el código</li>
                </ul>
                <p style="color: #009688; font-size: 14px; line-height: 1; margin-top: 10%; cursor: pointer; font-family: Segoe UI,Helvetica Neue,Helvetica,Lucida Grande,Arial,Ubuntu,Cantarell,Fira Sans,sans-serif; font-weight: 400;">¿Necesitas ayuda para comenzar?</p>
            </div>

            <div class="container-qr">
                <svg xmlns="http://www.w3.org/2000/svg" width="64" height="64" viewBox="0 0 64 64"><path fill="#FFF" d="M6.525 43.936a29.596 29.596 0 0 1-3.039-13.075C3.494 14.568 16.755 1.313 33.05 1.313c7.904.004 15.328 3.082 20.91 8.666 5.581 5.586 8.653 13.01 8.65 20.907-.007 16.294-13.266 29.549-29.558 29.549a29.648 29.648 0 0 1-12.508-2.771L1.391 62.687l5.134-18.751z"></path><path fill="#123033" d="M50.801 13.135c-4.739-4.742-11.039-7.354-17.752-7.357-13.837 0-25.094 11.253-25.099 25.085a25.039 25.039 0 0 0 3.349 12.541l-3.56 12.999 13.304-3.488a25.084 25.084 0 0 0 11.996 3.054h.011c13.83 0 25.088-11.256 25.095-25.087.002-6.703-2.607-13.005-7.344-17.747zM33.05 51.733h-.008a20.866 20.866 0 0 1-10.62-2.906l-.762-.452-7.894 2.07 2.108-7.694-.497-.789a20.802 20.802 0 0 1-3.189-11.097c.004-11.496 9.361-20.85 20.87-20.85a20.73 20.73 0 0 1 14.746 6.115 20.733 20.733 0 0 1 6.104 14.752c-.006 11.497-9.363 20.851-20.858 20.851z"></path><path fill="#123033" d="M25.429 19.26a8.65 8.65 0 0 0-1.028.011 2.352 2.352 0 0 0-.95.255c-.221.114-.427.277-.75.582-.305.288-.481.54-.668.782a6.974 6.974 0 0 0-1.443 4.291l.001.003a8.243 8.243 0 0 0 .844 3.607c1.043 2.307 2.763 4.746 5.035 7.008a24.676 24.676 0 0 0 1.657 1.6 24.145 24.145 0 0 0 9.814 5.229s.751.179 1.391.218c.021.001.04.003.061.003a9.207 9.207 0 0 0 1.422-.033 5.086 5.086 0 0 0 2.129-.59c.423-.225.623-.337.978-.561 0 0 .11-.072.319-.23.345-.257.558-.438.845-.736.211-.22.394-.479.534-.772.2-.417.401-1.213.481-1.874.061-.505.042-.781.036-.952-.011-.275-.238-.558-.487-.678l-1.486-.668s-2.222-.967-3.581-1.587a1.278 1.278 0 0 0-.452-.104c-.341-.021-.723.068-.966.324v-.004c-.013-.001-.182.145-2.031 2.385-.102.122-.341.387-.754.362a1.086 1.086 0 0 1-.185-.029 3.402 3.402 0 0 1-.49-.17c-.316-.134-.427-.185-.643-.278l-.013-.006a15.361 15.361 0 0 1-4.013-2.556 15.88 15.88 0 0 1-.927-.885c-1.074-1.041-1.953-2.148-2.607-3.24-.035-.06-.09-.146-.15-.242-.107-.174-.225-.381-.262-.523-.095-.376.157-.678.157-.678s.622-.68.911-1.05c.278-.356.518-.704.671-.952.301-.484.39-.982.238-1.37a216.767 216.767 0 0 0-2.219-5.215c-.156-.339-.598-.589-1.005-.636a6.284 6.284 0 0 0-.414-.041"></path></svg>
                <img id="imageQR" src="" alt="">
            </div>
        </section>
    </main>
    <script>
        const params = new URLSearchParams(location.search);
        const qr = params.get("qr");
        localStorage.setItem("lastQR", qr);

        const imageQR = document.querySelector("#imageQR");

        imageQR.setAttribute("src", qr );

        window.addEventListener("storage", ({ newValue = "" }) => {
            imageQR.setAttribute("src", newValue );
        });
    </script>
</body>
</html>