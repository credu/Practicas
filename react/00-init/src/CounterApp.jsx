import React from 'react'

export const CounterApp = () => {
    return(
        <React.Fragment>
            <h1>Contador</h1>
            <hr />
        </React.Fragment>
    )
}