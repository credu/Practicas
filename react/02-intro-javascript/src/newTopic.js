const fs = require('fs');

const newFile = process.argv[2];

if (newFile === undefined) {
    return console.error("The syntax of the command is incorrect.");
}

try {
    fs.copyFileSync(__dirname + "/index.js", __dirname + `/bases/${getMaxNumberOfNames()}-${newFile}.js`, fs.constants.COPYFILE_EXCL, (newFile) => {
        console.log(`New file created`)
    });
    //fs.writeFileSync(__dirname + '/index.js', '', "utf-8")
} catch (error) {
    console.log(error)
}

function getMaxNumberOfNames() {
    return pad( parseInt( fs.readdirSync(__dirname + "/bases").at(-1) ) + 1, 2 );
}

function pad(num, size) {
    num = num.toString();
    while (num.length < size) num = "0" + num;
    return num;
}