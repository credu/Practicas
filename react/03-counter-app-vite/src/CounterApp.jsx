import PropTypes from 'prop-types'
import React, { useState } from 'react'

export const CounterApp = ({ value }) => {


  const [array, setArray] = useState(["elias","jesus"]);

  const [counter, setCounter] = useState(value);

  const handleAdd = () => {
    setCounter( counter + 1 );
  }

  const handleSubstract = () => {
    setCounter( counter - 1);
  }

  const handleReset = () => {
    setCounter( value );
  }

  const sumale = () => {
    setArray( [...array, "Vegetta"] )
  }

  return (
    <>
        <h1>CounterApp</h1>
        <h2> { counter } </h2>
        <h2> { array } </h2>
        <button onClick={ sumale }>sumar</button>
        <button onClick={ handleAdd }>+1</button>
        <button onClick={ handleSubstract }>-1</button>
        <button aria-label='btn-reset' onClick={ handleReset }>Reset</button>
    </>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
}