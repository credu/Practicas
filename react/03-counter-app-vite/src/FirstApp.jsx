import PropTypes from 'prop-types'

export const FirstApp = ({ title, subtitle = "Soy un subtitulo" }) => {

  // if (!title) {
  //   throw new Error("El title no existe");
  // }

  return (
    <>
      <h1 data-testid="test-title">{ title }</h1>
      <p>{ subtitle }</p>
      <p>{ subtitle }</p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
}

FirstApp.defaultProps = {
  //title: "Titulo",
  subtitle: "Soy un clasico"
}