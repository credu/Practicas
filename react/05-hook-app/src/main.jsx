import React from 'react'
import ReactDOM from 'react-dom/client'
import { BrowserRouter } from "react-router-dom";

import './index.css'

// import { Padre } from './07-tarea-memo/Padre'
// import { TodoApp } from './08-useReducer/TodoApp'
// import { MultipleCustomHooks } from './03-examples'
import { MainApp } from './09-useContext/MainApp'
import { Layout } from './05-useLayoutEffect/Layout';

ReactDOM.createRoot(document.getElementById('root')).render(
  <BrowserRouter>
    {/* <React.StrictMode> */}
      <MainApp />
    {/* </React.StrictMode> */}
  </BrowserRouter>
  
)
