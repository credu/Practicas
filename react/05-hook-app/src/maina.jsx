import React from 'react'
import ReactDOM from 'react-dom/client'

import './index.css'

import { MainApp } from './09-useContext/MainApp'
import { CounterApp } from './01-useState/CounterApp'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <MainApp />
  </React.StrictMode>
)
