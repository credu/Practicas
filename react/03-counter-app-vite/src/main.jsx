import React from 'react';
import ReactDOM from 'react-dom/client';
import { CounterApp } from './CounterApp';
import { FirstApp } from './FirstApp';
import { HelloWorldApp } from './HelloWorldApp'

import "./style.css"

ReactDOM.createRoot( document.getElementById("root") ).render(
    <React.StrictMode>
        <CounterApp value={ 900 } />
        
        {/* <FirstApp  /> */}
    </React.StrictMode>
); 