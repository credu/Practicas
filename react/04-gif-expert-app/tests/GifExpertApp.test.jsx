import { fireEvent, render, screen } from "@testing-library/react"
import { GifExpertApp } from "../src/GifExpertApp"

describe('Pruebas en GiftExpertApp', () => {

    test('Snapshot', () => {
        const { container } = render( <GifExpertApp /> );

        expect( container ).toMatchSnapshot();
    })
})