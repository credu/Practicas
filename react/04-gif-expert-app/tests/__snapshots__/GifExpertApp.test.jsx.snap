// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GiftExpertApp Snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
    autocomplete="off"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <div
    class="card-grid animate__animated animate__fadeIn animate__slow"
  >
    <h3>
      One Punch
    </h3>
    <h2
      class="animate__animated animate__flash animate__infinite animate__delay-5s"
    >
      Cargando...
    </h2>
    <button>
      X
    </button>
  </div>
</div>
`;
